### PR TITLE
Shorten mangled names

### PR DESCRIPTION
**Pull request by jesyspa**
Our current mangled names are inconveniently long.  I'm working on making them shorter, but it's quite a bit of work; this should make things a bit easier to read without reworking too much.

Assigning both of you as reviewers so you can share your thoughts on the readability; there's no behavioural changes here.

### CODE REVIEW

**Comment by GrigoriiSolnyshkin** [comment]
By the way, there is a fun issue that I think we should consider at some point. Kotlin is actually allowed to generate properties with the same name in the same scope although the developer can't do it manually. For example, it desugars  `x++` to something like `val <unary> = x; x = x + 1; <unary>`. If there is two increments, two `val <unary>`s are generated.
Something similar happens when desugaring `val (x, y) = ...` (a fake property with some fixed name for `(x, y)` is generated) and when using `_` as well.
Maybe it is worth to enumerate all the entities with a given name inside the scope.

File: plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt
Comment (GrigoriiSolnyshkin) [inline plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt:184]: Why do we want this `TF` if we already have `F` before in functions? It feels like after introduced abbreviations this `T` is less readable than `take` before.

**Review by GrigoriiSolnyshkin — CHANGES_REQUESTED** [review]
IMO abbreviations seem readable. I would probably stick with lowercase letters between `$` because it would be easier to spot them.

In the future I suppose we should document when we use `$` and `_` as separators and maybe introduce some new ones (as well as some parentheses-like separators which we would use some complex names inside other names, e.g. functions).

File: plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt
Comment (jesyspa) [reply plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt:182]: I was actually thinking something like `fun foo(a: (Int) -> Unit)` vs `fun foo(a: () -> Unit, b: Int)`, but I didn't check.

File: plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt
Comment (jesyspa) [reply plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt:184]: Because this is a type, so this makes it consistent with our other types like `T_Int`.  That said, I'm not entirely happy with this either...

**Comment by jesyspa** [comment]
> IMO abbreviations seem readable. I would probably stick with lowercase letters between `$` because it would be easier to spot them.

Hmm, that's an interesting one.  I feel like the capital letters make it obvious that we're dealing with an abbreviation.  But if we make the names more regular, that should be easier to tell anyway.
 
> In the future I suppose we should document when we use `$` and `_` as separators and maybe introduce some new ones (as well as some parentheses-like separators which we would use some complex names inside other names, e.g. functions).

I agree.  I think we use `_` when possible because `$` looks ugly, but `_` can collide because it's allowed in Kotlin names.  (On the other hand, so is `$` if you escape the name... maybe we should just sanitize it from underscores and stick mostly to underscores?)

File: plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/names/NameMangling.md
Comment (jesyspa) [reply plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/names/NameMangling.md:26]: We don't currently tag extension functions in any particular way.  I'd also personally prefer putting that tag on the type of the parameter, not the function name itself.

**Review by GrigoriiSolnyshkin — APPROVED** [review]
LGTM
I think it's worth to make issues regarding
1. names of function types
2. careful approach to separators
though

**Review by francescoo22 — APPROVED** [review]

**Comment by jesyspa** [comment]
Discussed with Francesco: the way we handle predicate names (and class names in general) here is unfortunate.  I'll figure out some better way before merging.

**Comment by jesyspa** [comment]
I have some ideas on how to get things to work better, but they rely on names being a bit less ad-hoc, which is a bigger change.  Still, I think it's worth doing.  I'll try to get it done this week.

**Comment by jesyspa** [comment]
I think I fixed the issue discussed with @francescoo22 , but now this is blocked on https://github.com/jesyspa/kotlin/pull/230

**Pull request merged**
